### PR TITLE
Api doc - Work in progress

### PR DESCRIPTION
**Pull request by brauliodiez**
Just an starting point, will review and decline to move forward but just to establish a checkpoint

### CODE REVIEW

File: api/api.mdx [hunk 1]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
 ```
 
-## FieldValidation
+### RecordValidationResults
 
-```typescript
-interface FieldValidationFunction {
-  (value: any, vm: any, customParams: any): ValidationResultSyncAsync;
-}
+Whenever we validate all form record entries (firedRecordValidations),
```
Comment (Nasdan) [inline api/api.mdx]: Replace `firedRecordValidations` by `validateRecords`.

Right now, `validateRecords` is the current name method that we are exporting as public

File: api/api.mdx [hunk 2]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
+a \_RecordValidationResult is returned.
```
Comment (Nasdan) [inline api/api.mdx]: What is the `\_`?

File: api/api.mdx [hunk 3]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
 
-interface FieldValidationFull {
-  validator: FieldValidationFunction;
-  customArgs?: object;
-  errorMessage: string | string[];
-}
+- **succeeded**: Wether if the record validations succeeded (true) or failed (false).
```
Comment (Nasdan) [inline api/api.mdx]: Typo: `Whether`

File: api/api.mdx [hunk 4]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
+export interface RecordValidationResult {
+  succeeded: boolean;
+  recordErrors: { [recordId: string]: ValidationResult };
+}
 ```
 
-## FormValidationResult
+### FormValidationResult
+
+Whenever we validate a form _validateFullForm_ (trigger
```
Comment (Nasdan) [inline api/api.mdx]: Replace `validateFullForm` by `validateForm`

File: api/api.mdx [hunk 5]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
-  fieldErrors: { [key: string]: ValidationResult };
-  formGlobalErrors: Array<ValidationResult>;
+  fieldErrors: { [fieldId: string]: ValidationResult };
+  recordErrors: { [recordId: string]: ValidationResult };
+}
+```
+
+Properties:
+
+- **succeeded**: Wether if all validations succeeded (true) or any of them failed (false).
```
Comment (Nasdan) [inline api/api.mdx]: Same typo

File: api/api.mdx [hunk 6]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
+- **fieldErrors**: Collection of field / ValidationResults.
+- **recordErrors**: Collections of record / ValidatinResults.
```
Comment (Nasdan) [inline api/api.mdx]: typo `ValidationResults`

File: api/api.mdx [hunk 7]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
+Arguments
+
+- **fieldValidatorArgs**: We pass here information like id of the field
+  to validate, actual value, values and customArgs. See **internal link**
+  for more information.
+
+Return Value
+
+- **ValidationResult**: Wether if field validations applied to the selected field
+  succeeded or not (in casenot additional information to be provided like id of the validator that failed plus error message).
```
Comment (Nasdan) [inline api/api.mdx]: Typo: `casenot`

File: api/api.mdx [hunk 8]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
+
+Arguments
+
+- **fieldValidatorArgs**: We pass here information like id of the field
+  to validate, actual value, values and customArgs. See **internal link**
+  for more information.
+
+Return Value
+
+- **Promise<ValidationResult\>**: Async promise, once promise is resolved returns wether if field validations applied to the selected field
```
Comment (Nasdan) [inline api/api.mdx]: Typo: `whether`

File: api/api.mdx [hunk 9]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
+  succeeded or not (in casenot additional information to be provided like id of the validator that failed plus error message).
```
Comment (Nasdan) [inline api/api.mdx]: `casenot` is a typo, isn't it?

File: api/api.mdx [hunk 10]
```diff
@@ -7,60 +7,202 @@ menu: 5. api
...
+
+Arguments
+
+- **recordValidatorArgs**: We pass here information like id of the field
+  to validate, actual value, values and customArgs. See **internal link**
+  for more information.
+
+Return Value
+
+- **ValidationResult**: Wether if field validations applied to the selected field
```
Comment (Nasdan) [inline api/api.mdx]: Same typos

File: api/api.mdx (hunk 1 above)
Comment (brauliodiez) [reply api/api.mdx]: It should be _ValidateRecord_ we cannot use plural we have a single record, I think we should update it on the code, do you agree?

File: api/api.mdx (hunk 2 above)
Comment (brauliodiez) [reply api/api.mdx]: typo

File: api/api.mdx (hunk 3 above)
Comment (brauliodiez) [reply api/api.mdx]: thanks, done

File: api/api.mdx (hunk 4 above)
Comment (brauliodiez) [reply api/api.mdx]: done

File: api/api.mdx (hunk 6 above)
Comment (brauliodiez) [reply api/api.mdx]: done

File: api/api.mdx (hunk 7 above)
Comment (brauliodiez) [reply api/api.mdx]: solved, thanks

**Pull request merged**